src/hooks/useCheckProductionPipelinesAndControlPanels.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from 'react';
+import { CallbackHook } from './types';
+import { Production } from '../interfaces/production';
+import { ResourcesCompactPipelineResponse } from '../../types/ateliere-live';
+
+export function useCheckProductionPipelinesAndControlPanels(): CallbackHook<
+  (
+    production: Production,
+    pipelines: ResourcesCompactPipelineResponse[] | undefined
+  ) => Production
+> {
+  const [loading, setLoading] = useState(false);
+
+  const checkProductionPipelinesAndControlPanels = (
+    production: Production,
+    pipelines: ResourcesCompactPipelineResponse[] | undefined
+  ) => {
+    if (!production.production_settings) return production;
+    const productionPipelines = production.production_settings?.pipelines;
```

Comment:
the "?" is useless here since you already check if production_settings is truthy. Not a big deal but a good practice to have.